os/arch/arm/src/amebad/amebad_watchdog_lowerhalf.c

```diff
@@ -0,0 +1,482 @@
+		priv->start_tick = SYSTIMER_TickGet();	// Record time right after wdt start
+		priv->on_pause_duration = 0;	// Reset on_pause_duration
+	}
+
+	return OK;
+}
+
+/****************************************************************************
+ * Name: amebad_wdg_stop
+ *
+ * Description:
+ *   Stop the watchdog timer
+ *
+ * Input Parameters:
+ *   lower - A pointer the publicly visible representation of the "lower-half"
+ *           driver state structure.
+ *
+ * Returned Values:
+ *   Zero on success; a negated errno value on failure.
+ *
+ ****************************************************************************/
+static int amebad_wdg_stop(FAR struct watchdog_lowerhalf_s *lower)
+{
+	struct amebad_wdg_lowerhalf_s *priv = (struct amebad_wdg_lowerhalf_s *)lower;
+	DEBUGASSERT(priv);
+	if (!priv) {
+		return -ENODEV;
+	}
+
+	/* Stop watchdog timer */
+	watchdog_stop();
+	priv->wdt_status = WDT_STOP;
+
+	return OK;
+}
+
+/****************************************************************************
+ * Name: amebad_wdg_keepalive
+ *
+ * Description:
+ *   Reset the watchdog timer to the current timeout value, prevent any
+ *   imminent watchdog timeouts.  This is sometimes referred as "pinging"
+ *   the watchdog timer or "petting the dog".
+ *
+ * Input Parameters:
+ *   lower - A pointer the publicly visible representation of the "lower-half"
+ *           driver state structure.
+ *
+ * Returned Values:
+ *   Zero on success; a negated errno value on failure.
+ *
+ ****************************************************************************/
+static int amebad_wdg_keepalive(FAR struct watchdog_lowerhalf_s *lower)
+{
+	struct amebad_wdg_lowerhalf_s *priv = (struct amebad_wdg_lowerhalf_s *)lower;
+	DEBUGASSERT(priv);
+	if (!priv) {
+		return -ENODEV;
+	}
+	if (priv->wdt_status == WDT_START) {
+		/* Petting the dog */
+		watchdog_refresh();
+
+		/* Update start time and reset on_pause_duration */
+		priv->start_tick = SYSTIMER_TickGet();
+		priv->on_pause_duration = 0;
+	}
+
+	return OK;
+}
+
+/****************************************************************************
+ * Name: amebad_wdg_getstatus
+ *
+ * Description:
+ *   Get the current watchdog timer status
+ *
+ * Input Parameters:
+ *   lower  - A pointer the publicly visible representation of the "lower-half"
+ *            driver state structure.
+ *   status - The location to return the watchdog status information.
+ *
+ * Returned Values:
+ *   Zero on success; a negated errno value on failure.
+ *
+ ****************************************************************************/
+static int amebad_wdg_getstatus(FAR struct watchdog_lowerhalf_s *lower, FAR struct watchdog_status_s *status)
+{
+	struct amebad_wdg_lowerhalf_s *priv = (struct amebad_wdg_lowerhalf_s *)lower;
+	DEBUGASSERT(priv);
+	if (!priv) {
+		return -ENODEV;
+	}
+
+	/* Reture watchdog flag */
+	status->flags = WDFLAGS_RESET;
+
+	/* Return ACTIVE if watchdog timer is started */
+	if (priv->wdt_status == WDT_START) {
```

Comment:
```
if (priv->wdt_status == ON_PAUSE) {
     status->timeleft = priv->timeout_left;
     status->flag |= WDFLAG_WAIT;
} else {
     ...
     status->timeout = priv->timeout_ms;
     status->timeleft = status->timeout - timeelapse;
     if (priv->wdt_status == WDT_START) {
         status->flag |= WDFLAG_ACTIVE;
     }
}

if (priv->intmode). ..

Reply:
The status->timeout also need to assign value in ON_PAUSE status actually.
I get what you mean, I will re-arrange that part.